src/Symfony/Component/DependencyInjection/Compiler/RegisterEnvVarProcessorsPass.php

```diff
@@ -43,14 +44,23 @@ public function process(ContainerBuilder $container)
                 foreach ($class::getProvidedTypes() as $prefix => $type) {
                     $processors[$prefix] = new ServiceClosureArgument(new Reference($id));
                     $types[$prefix] = self::validateProvidedTypes($type, $class);
+
+                    if (!$types[$prefix]) {
```

Comment:
can this happen at all? explode always creates an array with at least one element, isn't it?

Reply:
nope :)